watertap/unit_models/tests/test_anaerobic_digester.py

```diff
@@ -206,4 +206,27 @@ def configure(self):
         self.unit_solutions[m.fs.unit.electricity_consumption[0]] = 23.7291667
         self.unit_solutions[m.fs.unit.hydraulic_retention_time[0]] = 1880470.588
 
+        # Conservation check
+        self.conservation_equality = {
+            "Inlet expression 1": m.fs.unit.inlet.flow_vol[0],
+            "Outlet expression 1": (
+                m.fs.unit.liquid_outlet.flow_vol[0] * m.fs.props.dens_mass
+                + m.fs.unit.vapor_outlet.flow_vol[0] * m.fs.props_vap.dens_mass
+            )
+            / m.fs.props.dens_mass,
+            "Inlet expression 2": (
+                m.fs.unit.inlet.flow_vol[0]
+                * m.fs.props.dens_mass
+                * m.fs.props.cp_mass
+                * (m.fs.unit.inlet.temperature[0] - m.fs.props.temperature_ref)
+            )
+            - (
+                m.fs.unit.liquid_outlet.flow_vol[0]
+                * m.fs.props.dens_mass
+                * m.fs.props.cp_mass
+                * (m.fs.unit.liquid_outlet.temperature[0] - m.fs.props.temperature_ref)
+            ),
+            "Outlet expression 2": -1 * m.fs.unit.liquid_phase.enthalpy_transfer[0],
+        }
```

Comment:
I think it would be better to nest the related expressions, like the following:
```
conservation_check = {
   "check 1": [expression_in, expression_out],
   "check 2": [expression_in, expression_out],
}
```
Or if we think it would be good to be explicit with a key for "in" and "out":
```
conservation_check = {
   "check 1": {"in": expression_in, "out": expression_out},
   "check 2": {"in": expression_in, "out": expression_out},
}
```

This way we don't have to require the user to provide the exact string like "Outlet expression X". The check can just iterate through the highest key (in my example its "check 1" and "check 2", but those names could be anything), and the first example will just make sure that the dictionary results in a 2 item list and know those things should be checked to be approximately equal, or the second example requires the "in" and "out" hard coded but that's not too bad.

Reply:
My one concern with this is that it might come across looking a bit too clunky. The current implementation is also a bit clunky imo, but I think it looks better than this:
![image](https://github.com/watertap-org/watertap/assets/96305519/967a5e0e-c203-4f5b-9555-471cffe77ba1)

Reply:
Tim's 2nd suggestion seems the least error prone to me and is the most explicit.

Just to throw in another curveball, I don't see any reason we could just use Pyomo's expression system itself. e.g.
```
"Check 1": m.fs.unit.inlet.flow_vol[0] == m.fs.unit.outlet.flow_vol[0],
```

This is a bit more complicated on the back end, but not overly so.